readthedocs/api/v3/serializers.py

```diff
@@ -0,0 +1,472 @@
+
+    This serializer validates that no field is sent at all in the request.
+    """
+
+    class Meta:
+        model = Build
+        fields = []
+
+
+class BuildLinksSerializer(BaseLinksSerializer):
+    _self = serializers.SerializerMethodField()
+    version = serializers.SerializerMethodField()
+    project = serializers.SerializerMethodField()
+
+    def get__self(self, obj):
+        path = reverse(
+            'projects-builds-detail',
+            kwargs={
+                'parent_lookup_project__slug': obj.project.slug,
+                'build_pk': obj.pk,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_version(self, obj):
+        path = reverse(
+            'projects-versions-detail',
+            kwargs={
+                'parent_lookup_project__slug': obj.project.slug,
+                'version_slug': obj.version.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_project(self, obj):
+        path = reverse(
+            'projects-detail',
+            kwargs={
+                'project_slug': obj.project.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+
+class BuildConfigSerializer(FlexFieldsSerializerMixin, serializers.Serializer):
+
+    """
+    Render ``Build.config`` property without modifying it.
+
+    .. note::
+
+       Any change on the output of that property will be reflected here,
+       which may produce incompatible changes in the API.
+    """
+
+    def to_representation(self, instance):
+        # For now, we want to return the ``config`` object as it is without
+        # manipulating it.
+        return instance
+
+
+class BuildStateSerializer(serializers.Serializer):
+    code = serializers.CharField(source='state')
+    name = serializers.SerializerMethodField()
+
+    def get_name(self, obj):
+        return obj.state.title()
+
+
+class BuildSerializer(FlexFieldsModelSerializer):
+
+    project = serializers.SlugRelatedField(slug_field='slug', read_only=True)
+    version = serializers.SlugRelatedField(slug_field='slug', read_only=True)
+    created = serializers.DateTimeField(source='date')
+    finished = serializers.SerializerMethodField()
+    success = serializers.SerializerMethodField()
+    duration = serializers.IntegerField(source='length')
+    state = BuildStateSerializer(source='*')
+    links = BuildLinksSerializer(source='*')
+
+    expandable_fields = dict(
+        config=(
+            BuildConfigSerializer,
+            dict(
+                source='config',
+            ),
+        ),
+    )
+
+    class Meta:
+        model = Build
+        fields = [
+            'id',
+            'version',
+            'project',
+            'created',
+            'finished',
+            'duration',
+            'state',
+            'success',
+            'error',
+            'commit',
+            'builder',
+            'cold_storage',
```

Comment:
We are leaking builder and cold_storage to common users, we only show this to admin users in v2

Reply:
I agree that `cold_storage` could be removed from here, considering that we are going to have everything in cold storage in the near future. So, this won't be useful anymore.

`builder` is not a problem, but you are right that this should probably be only for admins. I will remove it for now, and we can consider re-add it later if we want to expose different things to different users. Also, if we are going to use lambda functions or similar, the `builder` won't make sense anymore.